perf-dashboard/deployment-latency/tests/CollectDeploymentLatencyTest.php

```diff
@@ -58,49 +57,66 @@ public function testDeploymentLatency()
         $phpVersions = [
             '5.6',
             '7.0',
-            '7.1'
+            '7.1',
+            '7.2'
         ];
         $types = [
-            'xrt',
-            'builder'
+            'debian8',
+            'ubuntu16'
         ];
         $gcloudTrack = getenv('GCLOUD_TRACK') === 'beta' ? 'beta' : '';
         foreach ($phpVersions as $phpVersion) {
-            $dir = self::createApp($phpVersion);
-            chdir($dir);
             foreach ($types as $type) {
-                $reportName = sprintf('%s-%s', $type, $phpVersion);
-                $failureCount = 0;
-                $command = 'gcloud -q ' . $gcloudTrack . ' app deploy'
-                    . ' --version '
-                    . str_replace('.', '', $reportName)
-                    . ' --no-stop-previous-version --no-promote';
-                $configCmd = 'gcloud config set app/use_runtime_builders '
-                    . ($type === 'xrt' ? 'false' : 'true');
-                self::execWithError($configCmd, 'runtime-builders-config');
-                $latency = 0.0;
-                while ($failureCount < self::DEPLOYMENT_MAX_RETRY) {
-                    $start = microtime(true);
-                    $ret = self::execWithResult($command);
-                    if ($ret === 0) {
-                        $latency = microtime(true) - $start;
-                        break;
+                $dir = self::createApp($phpVersion);
+                chdir($dir);
+                try {
+                    $reportName = sprintf('%s-%s', $type, $phpVersion);
+                    $failureCount = 0;
+                    $command = 'gcloud -q ' . $gcloudTrack . ' app deploy'
+                        . ' --version '
+                        . str_replace('.', '', $reportName)
+                        . ' --no-stop-previous-version --no-promote';
+                    $configCmd = 'gcloud config set app/use_runtime_builders true';
+                    self::execWithError($configCmd, 'runtime-builders-config');
+                    $f = fopen('app.yaml', 'a+');
```

Comment:
won't there be two `base_os` lines in the second run through if we're creating the app outside of the `$types` foreach?

Reply:
We're creating the app inside of the foreach now